backend/authentication/auth_controller.js

```diff
@@ -0,0 +1,140 @@
+const axios = require("axios");
+const jwt = require("jsonwebtoken");
+const prisma = require("../prisma_client");
+const bcrypt = require("bcrypt");
+require("dotenv").config();
+
+const FRONTEND_URL = process.env.FRONTEND_URL;
+const BACKEND_URL = process.env.BACKEND_URL;
+const JWT_SECRET = process.env.JWT_SECRET;
+const SALT_ROUNDS = 10;
+
+exports.register = async (req, res) => {
+  const { username, email, password, fullName } = req.body;
```

Comment:
I would add checking that each of the required variables exist

Reply:
Done.